addons/gloot/ctrl_inventory_grid.gd

```diff
@@ -314,18 +334,62 @@ func _select(item: InventoryItem) -> void:
     _selected_item = item
     _refresh_selection()
 
+func switch_virtual_cursor_grid(new_grid):
+    if not _grabbed_with_virtual_cursor or not new_grid:
+        return
+
+    _virtual_cursor_gird = new_grid
+    _virtual_cursor = new_grid.rect_global_position
+
 
 func _input(event: InputEvent) -> void:
-    if !(event is InputEventMouseButton):
+    if event.is_action_released("ui_cancel"):
+        _cancel_grab()
         return
 
-    if !_grabbed_ctrl_inventory_item:
+    _handle_virtual_cursor(event)
+    _handle_mouse(event)
+
+func _handle_virtual_cursor(event: InputEvent):
+    if not _grabbed_with_virtual_cursor:
+        return
+
+    var delta
+    if event.is_action_pressed("ui_left"):
+        delta = Vector2(-_virtual_cursor_gird.field_dimensions.x, 0)
+    if event.is_action_pressed("ui_right"):
+        delta = Vector2(_virtual_cursor_gird.field_dimensions.x, 0)
+    if event.is_action_pressed("ui_up"):
+        delta = Vector2(0, -_virtual_cursor_gird.field_dimensions.y)
+    if event.is_action_pressed("ui_down"):
+        delta = Vector2(0, _virtual_cursor_gird.field_dimensions.y)
+    if event.is_action_released("ui_accept"):
+        if _virtual_cursor_can_drop:
+            _drop_item()
+        else:
+            _virtual_cursor_can_drop = true
+
+    if delta:
+        var new_pos = _virtual_cursor + delta
+        if _virtual_cursor_gird.get_global_rect().has_point(new_pos):
+            _virtual_cursor = new_pos
+        else:
+            emit_signal("virtual_cursor_moved_outside", _virtual_cursor_gird, delta.normalized())
```

Comment:
I think using an enum for the second argument instead of a `Vector2` would be more convenient. We only have 4 possible values here after all (up, down, left and right).